### DISCUSSION TITLE
Does logto support grant_type=password in OAuth2 ?

### DISCUSSION THREAD

**Discussion by zhfkt** [logto-io/logto#5638] [Q&A]
Does logto support `grant_type=password` in OAuth2 ? I only notice `grant_type=authorization_code` currently.

**Comment by xiaoyijun** [comment]
Hi @zhfkt , you can checkout any tenant's oidc config endpoint to see supported grant types (e.g., https://default.logto.app/oidc/.well-known/openid-configuration) and check the `grant_types_supported` field:
<img width="264" alt="image" src="https://github.com/logto-io/logto/assets/10806653/366ab348-8d5f-4925-a96c-edc05e794717">

`password` grant type is not supported, and could you please tell me why you need this grant type?

**Reply by zhfkt** [reply]
Because I would like to use my own login page. As discussed in https://stackoverflow.com/questions/39356300/avoid-keycloak-default-login-page-and-use-project-login-page, we could implement it by `grant_type=password`. From, https://github.com/logto-io/logto/discussions/4676, I know we are able to customize the CSS of the experience pages on the sign-in experience page. But I worried it is not enough since modifying CSS is just polishing the login page instead of creating our own page.

**Reply by xiaoyijun** [reply]
Thank you for your feedback.
We're currently working on the feature of customization of the login interface, but it might not be implemented through the `grant_type=password` method. The best approach at the moment is to refer to our experience source code to create your own login page.

**Reply by zhfkt** [reply]
Thanks
